.github/actions/publish-npm-package/action.yml

```diff
@@ -0,0 +1,134 @@
+      run: npm install -g npm@latest
+
+    - name: Install node dependencies
+      shell: bash
+      run: yarn install --immutable
+
+    - name: Build package
+      id: build
+      working-directory: packages/react-native-gesture-handler
+      shell: bash
+      run: npm pack
+
+    - name: Add package name to env
+      working-directory: packages/react-native-gesture-handler
+      shell: bash
+      run: echo "PACKAGE_NAME=$(ls -l | egrep -o "react-native-gesture-handler-(.*)(=?\.tgz)")" >> $GITHUB_ENV
+
+    - name: Assert PACKAGE_NAME
+      if: ${{ env.PACKAGE_NAME == 'PLACEHOLDER' }}
+      shell: bash
+      run: exit 1 # If we end up here, it means that package was not generated.
+
+    - name: Upload npm package to GitHub
+      uses: actions/upload-artifact@v4
+      with:
+        name: ${{ env.PACKAGE_NAME }}
+        path: './packages/react-native-gesture-handler/${{ env.PACKAGE_NAME }}'
+
+    - name: Figure out the correct npm tag
+      id: figure-out-npm-tag
+      shell: bash
+      run: |
+        if [ "${{ inputs.is-commitly }}" = "true" ]; then
+          TAG_ARGUMENT="--tag nightly"
+        else
+          if [ $(node ./scripts/release/should-be-latest.js ${{ steps.set-package-version.outputs.version }}) = "true" ]; then
+            TAG_ARGUMENT="--tag latest"
+          else
+            TAG_ARGUMENT=""
+          fi
+        fi
+
+        echo "Determined tag argument: $TAG_ARGUMENT"
+        echo "tag-argument=$TAG_ARGUMENT" >> $GITHUB_OUTPUT
+
+    - name: Print outputs
+      shell: bash
+      run: |
+        echo "Version to be published: ${{ steps.set-package-version.outputs.version }}"
+        echo "NPM tag argument: ${{ steps.figure-out-npm-tag.outputs.tag-argument }}"
+
+    - name: Publish npm package
+      shell: bash
+      if: inputs.dry-run == 'false'
+      working-directory: packages/react-native-gesture-handler
+      run: |
+        npm publish $PACKAGE_NAME --provenance ${{ steps.figure-out-npm-tag.outputs.tag-argument }}
+
+    - name: Don't publish if dry-run is true
+      shell: bash
+      if: inputs.dry-run == 'true'
+      run: |
+        echo "At this point, the following command would have been run: npm publish $PACKAGE_NAME --provenance ${{ steps.figure-out-npm-tag.outputs.tag-argument }}"
+
+    - name: Configure git
+      if: inputs.is-commitly == 'false'
+      shell: bash
+      run: |
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+
+    - name: Create release commit
+      if: inputs.is-commitly == 'false'
+      shell: bash
+      run: |
+        git add packages/react-native-gesture-handler/package.json
+        git commit -m "Release v${{ steps.set-package-version.outputs.version }}"
+        
+        if [ "${{ inputs.dry-run }}" = 'false' ]; then
+          echo "Pushing release commit to origin..."
+          git push
+        else
+          echo "Dry run mode: skipping git push"
+        fi
```

Comment:
Right now our release commits also update `Podfile.locks`. I don't think it is a problem, I just wanted to mention that 😅

Reply:
I'm not sure if it matters that much, those are only in the test apps so they don't end up in the package anyway.

We may even consider not versioning on the `main` branch and set it to `0.0.0` or something like that.